packages/rum-core/src/performance-monitoring/transaction-service.js

```diff
@@ -352,6 +355,43 @@ class TransactionService {
     )
   }
 
+  setSession(tr) {
```

Comment:
this function is super confusing. can you change this a bit?.

Reply:
@vigneshshanmugam , this is not a very useful comment, I'm happy to make this function more clear, but you need to provide more detail on what exactly do you mean here.

Reply:
My bad, forgot to post the alternative snippet, was thinking to reduce the conditionals in this way or similar pattern. 

```js
setSession(tr) {
    let session = this._config.get("session");
    let value = {};
    if (
      typeof session == "boolean" ||
      (session.timestamp && Date.now() - session.timestamp > SESSION_TIMEOUT)
    ) {
      value = {
        id: generateRandomId(16),
        sequence: 1,
      };
    } else {
      value = {
        id: session.id,
        sequence: session.sequence ? session.sequence + 1 : 1,
      };
    }
    tr.session = value;

    const sessionConfig = {
      session: { timestamp: Date.now(), ...value },
    };
    this._config.setConfig(sessionConfig);
    this._config.setLocalConfig(sessionConfig, true);
  }

```

Lets not do this now, May be this is going to change after our discussion.